pkg/operator/target_config_reconciler.go

```diff
@@ -90,12 +99,54 @@ func NewTargetConfigReconciler(
 		queue:               workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "TargetConfigReconciler"),
 		excludedNamespaces:  excludedNamespaces,
 		targetImagePullSpec: targetImagePullSpec,
+		apiExtensionsClient: apiExtensionsClient,
+		migrationsClient:    migrationsClient,
 	}
 	operatorClientInformer.Informer().AddEventHandler(c.eventHandler())
 
 	return c
 }
 
+// (4.8 only): Perform CRD v1beta1->v1 migration and update stored versions
+// See https://bugzilla.redhat.com/show_bug.cgi?id=1992478
+func (c TargetConfigReconciler) migrateStorageVersion() error {
+	existingMigration, err := c.migrationsClient.MigrationV1alpha1().StorageVersionMigrations().Get(c.ctx, "operator-kubedescheduler-storage-version-migration", metav1.GetOptions{})
+	// if the migration already exists, return nil
+	// if we get any error (besides IsNotFound) return the error
+	// otherwise, proceed
+	if (err != nil && !apierrors.IsNotFound(err)) || existingMigration != nil {
+		return err
+	}
+
+	// first, create the storage version migration to update existing objects to v1
+	migrationBytes := v410_00_assets.MustAsset("v4.1.0/kube-descheduler/v1-migration.yaml")
+	requiredObj, err := runtime.Decode(migrationscheme.Codecs.UniversalDecoder(migrationv1alpha1.SchemeGroupVersion), migrationBytes)
+	if err != nil {
+		return err
+	}
+	migration := requiredObj.(*migrationv1alpha1.StorageVersionMigration)
+	_, err = c.migrationsClient.MigrationV1alpha1().StorageVersionMigrations().Create(c.ctx, migration, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+	// now update the CRD itself to remove v1beta1 from storage version if necessary
+	crd, err := c.apiExtensionsClient.ApiextensionsV1().CustomResourceDefinitions().Get(c.ctx, "kubedeschedulers.operator.openshift.io", metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	storedVersions := sets.NewString(crd.Status.StoredVersions...)
+	if !storedVersions.Has("v1beta1") && storedVersions.Has("v1") {
+		// nothing to do
+		return nil
+	}
+	crd.Status.StoredVersions = []string{"v1"}
```

Comment:
Can you remove "v1beta1" from the list instead of replacing the whole list with ["v1"]? The semantics is to "remove v1beta1 from the list", not "replace the list of stored versions with v1".